localstack-core/localstack/services/lambda_/provider.py

```diff
@@ -1999,48 +2008,51 @@ def validate_event_source_mapping(self, context, request):
         else:
             fn_arn = api_utils.unqualified_lambda_arn(function_name, account, region)
 
-        def _get_mapping_sources(mapping: dict[str, Any]) -> list[str]:
-            if event_source_arn := mapping.get("EventSourceArn"):
-                return [event_source_arn]
-            return (
-                mapping.get("SelfManagedEventSource", {})
-                .get("Endpoints", {})
-                .get("KAFKA_BOOTSTRAP_SERVERS", [])
-            )
-
-        # check for event source duplicates
-        # TODO: currently validated for sqs, kinesis, and dynamodb
-        service_id = load_service(service).service_id
-        for uuid, mapping in state.event_source_mappings.items():
-            mapping_sources = _get_mapping_sources(mapping)
-            request_sources = _get_mapping_sources(request)
-            if mapping["FunctionArn"] == fn_arn and (
-                set(mapping_sources).intersection(request_sources)
-            ):
-                if service == "sqs":
-                    # *shakes fist at SQS*
-                    raise ResourceConflictException(
-                        f'An event source mapping with {service_id} arn (" {mapping["EventSourceArn"]} ") '
-                        f'and function (" {function_name} ") already exists. Please update or delete the '
-                        f"existing mapping with UUID {uuid}",
-                        Type="User",
-                    )
-                elif service == "kafka":
-                    if set(mapping["Topics"]).intersection(request["Topics"]):
+        # Check we are validating a CreateEventSourceMapping request
+        if is_create_esm_request:
```

Comment:
Only validate the below checks if the request is a `CreateEventSourceMapping` since this will check for duplicates etc.